### PR TITLE
remove fallback to indexdao for critical workflow.

### PR DESCRIPTION
**Pull request by manan164**
This PR contains following things.
1) Removed falling back to indexDAO for critical workflow like scheduling workflows
2) Only UI call back will be fall back to indexDAO.
3) Wrote unit test
4) Fix existing unit test.

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java [hunk 1]
```diff
@@ -71,24 +70,51 @@ public ExecutionDAOFacade(ExecutionDAO executionDAO, IndexDAO indexDAO, ObjectMa
...
+     * @param workflowId   the id of the workflow to be fetched
+     * @param includeTasks if true, fetches the {@link Task} data in the workflow.
+     * @return the {@link Workflow} object
+     * @throws ApplicationException if
+     *                              <ul>
+     *                              <li>no such {@link Workflow} is found</li>
+     *                              <li>parsing the {@link Workflow} object fails</li>
+     *                              </ul>
+     */
+    public Workflow getWorkflowByIdWithFallbackToIndexDAO(String workflowId, boolean includeTasks) {
```
Comment (naveenchlsn) [inline core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java]: name feels too long. What do you think about readWorkFlow ?

**Comment by naveenchlsn** [comment]
The dashboard also has fields for searching using name? Can you please verify that works as well.

**Comment by manan164** [comment]
Hi @mudit3774 I have addressed review comments.

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java (hunk 1 above)
Comment (manan164) [reply core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java]: Done.

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java [hunk 2]
```diff
@@ -71,24 +70,51 @@ public ExecutionDAOFacade(ExecutionDAO executionDAO, IndexDAO indexDAO, ObjectMa
     public Workflow getWorkflowById(String workflowId, boolean includeTasks) {
         Workflow workflow = executionDAO.getWorkflow(workflowId, includeTasks);
         if (workflow == null) {
-//            LOGGER.debug("Workflow {} not found in executionDAO, checking indexDAO", workflowId);
-//            String json = indexDAO.get(workflowId, RAW_JSON_FIELD);
-//            if (json == null) {
+            String errorMsg = String.format("No such workflow found by id: %s", workflowId);
```
[the remark this answers is not in the document]
Comment (manan164) [reply core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java]: Done.

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java [hunk 3]
```diff
@@ -71,24 +70,51 @@ public ExecutionDAOFacade(ExecutionDAO executionDAO, IndexDAO indexDAO, ObjectMa
...
+            LOGGER.error(errorMsg);
+            throw new ApplicationException(ApplicationException.Code.NOT_FOUND, errorMsg);
```
[the remark this answers is not in the document]
Comment (mudit3774) [reply core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java]: @manan164 can you please justify why a 404 is thrown?

File: core/src/test/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacadeTest.java [hunk 4]
```diff
@@ -77,6 +76,31 @@ public void tesGetWorkflowById() throws Exception {
...
+        String jsonString = new String(bytes);
+        when(indexDAO.get(any(), any())).thenReturn(jsonString);
+        workflow = executionDAOFacade.fetchWorkFlow("workflowId");
+        assertNotNull(workflow);
+        verify(indexDAO, times(1)).get(any(), any());
+    }
+
+    @Test (expected = java.lang.Exception.class)
+    public void readWorkflowFailure() throws Exception {
+
```
Comment (mudit3774) [inline core/src/test/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacadeTest.java]: Consider removing this new line

**Pull request merged**